packages/flutter_tools/test/commands/packages_test.dart

```diff
@@ -98,18 +101,40 @@ void main() {
       ProcessManager: () {
         return new MockProcessManager((List<dynamic> command) {
           log.add(command);
+          return new MockProcess();
+        });
+      },
+    });
+    testUsingContext('publish', () async {
+      log.clear();
+      await createTestCommandRunner(new PackagesCommand()).run(<String>['packages', 'pub', 'publish']);
+      expect(log, hasLength(1));
+      expect(log[0], hasLength(2));
+      expect(log[0][0], matches(r'dart-sdk[\\/]bin[\\/]pub'));
+      expect(log[0][1], 'publish');
+    }, overrides: <Type, Generator>{
+      ProcessManager: () {
+        return new MockProcessManager((List<dynamic> command) {
```

Comment:
As per our offline conversation, this test will end up writing on stdout. It would be nicer if pubInteractively somehow could use an alternative stdout/stderr and write to that. Then the test could also see that the stdout is as expected

Reply:
Done.